power_specs.py

```diff
@@ -15,683 +18,686 @@
-             base="/home/spb41/Lyman-alpha/MinParametricRecon/runs/",bf="best-fit/",suf="flux-power/", ext="_flux_power.txt"):
-                power_spec.__init__(self, Snaps,Zz,sdsskbins,knotpos, om, H0,box,base,suf, ext)
-                (k_bf,Pk_bf)= self.loadpk(bf+suf+"snapshot_000"+self.ext,self.bfbox)
-                ind=np.where(k_bf <= kmax)
-                self.kbins=k_bf[ind]
-
-        def plot_z(self,Sims,redshift,title="Relative Flux Power",ylabel=r"$\mathrm{P}_\mathrm{F}(k,p)\,/\,\mathrm{P}_\mathrm{F}(k,p_0)$", legend=True):
-                power_spec.plot_z(self,Sims,redshift,title,ylabel,legend)
-                if legend:
-                        kbins=self.GetSDSSkbins(redshift)
-                        plt.axvspan(kbins[0], kbins[-1], color="#B0B0B0")
-                plt.ylim(self.ymin,self.ymax)
-                plt.xlim(self.kbins[0]*0.8, 10)
-
-        """Get the kbins to interpolate onto"""
-        def Getkbins(self):
-               return self.kbins
-
-        """Load the SDSS power spectrum"""
-        def MacDonaldPF(self,sdss, zz):
-               psdss=sdss[np.where(sdss[:,0] == zz)][:,1:3]
-               fbar=math.exp(-0.0023*(1+zz)**3.65)
-               #multiply by the hubble parameter to be in 1/(km/s)
-               scale=self.Hubble(zz)/(1.0+zz)
-               PF=psdss[:,1]*fbar**2/scale
-               k=psdss[:,0]*scale
-               return (k, PF)
-
-        """Load a Pk. Different function due to needing to be different for each class"""
-        def loadpk(self, path,box):
-                #Adjust Fourier convention.
-                flux_power=np.loadtxt(self.base+path)
-                scale=self.H0/box
-                k=(flux_power[1:,0]-0.5)*scale*2.0*math.pi
-                PF=flux_power[1:,1]/scale
-                return (k, PF)
+    figprefix="/flux-figure"
+    kbins=np.array([])
+    def __init__(self, Snaps=("snapshot_000", "snapshot_001","snapshot_002","snapshot_003","snapshot_004","snapshot_005","snapshot_006","snapshot_007","snapshot_008","snapshot_009","snapshot_010","snapshot_011"),
+         Zz=np.array([4.2,4.0,3.8,3.6,3.4,3.2,3.0,2.8,2.6,2.4,2.2,2.0]),
+         sdsskbins=np.array([0.00141,0.00178,0.00224,0.00282,0.00355,0.00447,0.00562,0.00708,0.00891,0.01122,0.01413,0.01778]),
+         knotpos=np.array([0.07,0.15,0.475, 0.75, 1.19, 1.89,4,25]), om=0.266, H0=0.71,box=60.0,kmax=4.0,
+         base="/home/spb41/Lyman-alpha/MinParametricRecon/runs/",bf="best-fit/",suf="flux-power/", ext="_flux_power.txt"):
+        power_spec.__init__(self, Snaps,Zz,sdsskbins,knotpos, om, H0,box,base,suf, ext)
+        (k_bf,_Pk_bf)= self.loadpk(bf+suf+"snapshot_000"+self.ext,self.bfbox)
+        ind=np.where(k_bf <= kmax)
+        self.kbins=k_bf[ind]
+
+    def plot_z(self,Sims,redshift,title="Relative Flux Power",ylabel=r"$\mathrm{P}_\mathrm{F}(k,p)\,/\,\mathrm{P}_\mathrm{F}(k,p_0)$", legend=True):
```

Comment:
The solution to W0221 is to rename 'Sims' to 'Knot' in the function definition (or 'Knot' to 'Sims'). The reason this is a good idea is that if a caller passes arguments by keyword the child class will not look like the parent.